tooling/metamodel-generator/hibernate-jpamodelgen.gradle

```diff
@@ -43,6 +43,40 @@ compileTestJava {
 }
 
 
+// Tests with records
+if ( gradle.ext.javaVersions.test.release.asInt() >= 17 && gradle.ext.javaToolchainEnabled ) {
```

Comment:
Can you please add a separate source set like is for the hibernate-core module? I'll approve running the workflow, but I'm pretty sure the run will fail because of this change, since we require compatibility with Java 11